File: bindings/flutter/ios/kevy_anchor.c
// Link anchor for the dart:ffi door. flutter_kevy calls the engine
// through DynamicLibrary.process(), which resolves symbols at runtime —
// so at link time nothing references libkevy_ffi.a and the linker
// dead-strips it (Xcode: "Library 'kevy_ffi' not found"). This one
// forced reference to a single engine symbol pulls the static library's
// object in, keeping every kevy_* export in the app binary where
// DynamicLibrary.process() can find them. (The Swift/expo door needs no
// anchor: its Swift shell calls the symbols directly.)
#include <stdint.h>

extern uint32_t kevy_abi(void);

__attribute__((used, visibility("default"))) uint32_t
_kevy_flutter_link_anchor(void) {
    return kevy_abi();
}

File: bindings/flutter/ios/KevyAnchor.swift
// Link anchor for the dart:ffi door, matching the expo door's mechanism.
//
// flutter_kevy reaches the engine through Dart's DynamicLibrary.process(),
// which resolves symbols at runtime — so at link time nothing references
// the vendored Kevy.xcframework and CocoaPods neither links it nor keeps
// its symbols ("Library 'kevy_ffi' not found" at link). Importing the
// `Kevy` clang module (the same `import Kevy` the green expo door's Swift
// shell uses) is what makes CocoaPods link the xcframework; the forced
// reference to kevy_abi keeps its symbols in the app binary for
// DynamicLibrary.process() to find. A bare-C extern does not trigger the
// pod's module link — the module import does.
import Foundation
import Kevy

@objc public final class KevyLinkAnchor: NSObject {
    // Never called at runtime — its only job is to be a link-time
    // reference so the engine's symbols survive into the binary.
    @objc public static func anchor() -> UInt32 {
        return kevy_abi()
    }
}
